zig.html.markdown

```diff
@@ -0,0 +1,980 @@
+---
+language: "zig"
+filename: learnzig.zig
+contributors:
+    - ["Philippe Pittoli", "https://karchnu.fr/"]
+---
+
+
+[Zig][ziglang] aims to be a replacement for the C programming language.
+
+**WARNING**: this document expects you to understand a few basic concepts in computer science, such as pointers, stack and heap memory, etc.
+
+**WARNING**: Zig isn't considered as ready for production. Bugs are expected.
+DO NOT TRY ZIG AS YOUR FIRST PROGRAMMING EXPERIENCE.
+The compiler, even the language and its libraries aren't ready, yet.
+You've been warned.
+
+Prior knowledge of C is recommended.
+
+
+## Quick overview: Zig compared to C
+
+- Syntax is mostly the same, with some improvements (less ambiguity).
+- Zig introduces namespaces.
+- Try and catch mechanism, which is both convenient, efficient and optional.
+- Most of the C undefined behaviors (UBs) are fixed.
```

Comment:
s/fixed/removed in zig's safe compilation modes/

Reply:
Are they only removed in safe compilation mode? :thinking: